### PR TITLE
WIP: V8 Debugging with SourceLink

### PR DESCRIPTION
**Pull request by warrenbuckley**
A simple PR about debugging with SourceLink & how to set it up based from my notes in the PR that was submitted for the SourceLink in UmbracoCMS core

### CODE REVIEW

**Comment by warrenbuckley** [comment]
When you lovely PR team get a chance to review & fix up or move this to the correct location feel free to do so 😄 

@marcemarc @Jeavon @jmayntzhusen @sofietoft

File: Reference/Debugging/index.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+---
+versionFrom: 8.1.0
+---
+
+# Debugging
```
Comment (marcemarc) [inline Reference/Debugging/index.md]: wondering whether we should call the article 'Debugging with SourceLink' ?

we have a page called 'debugging' already, in getting started - code - debugging...

https://our.umbraco.com/documentation/Getting-Started/Code/Debugging/

and we could add at the footer of that page a link back to this reference page... ?

File: Reference/Debugging/index.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+In version Umbraco 8.1+ we have enabled a debugging feature from Microsoft & Visual Studio called SourceLink which allows you to step into the native Umbraco CMS source code.
```
Comment (marcemarc) [inline Reference/Debugging/index.md]: Wondering if this is better ?

Microsoft & Visual Studio have introduced a new debugging technology called 'SourceLink' that enables source code debugging of certain .NET assemblies from NuGet - in version Umbraco 8.1+ this feature has been enabled to allow developers to step into the native Umbraco CMS source code.

File: Reference/Debugging/index.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+## Enabling SourceLink in Visual Studio 2019
```
Comment (marcemarc) [inline Reference/Debugging/index.md]: Is this the same in vs2017?

File: Reference/Debugging/index.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+## What is SourceLink ?
+You can find details from Microsoft about SourceLink here:
+
+https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/sourcelink<br/>
+https://www.hanselman.com/blog/ExploringNETCoresSourceLinkSteppingIntoTheSourceCodeOfNuGetPackagesYouDontOwn.aspx<br/>
+https://github.com/dotnet/sourcelink
+
+## Verifying this works
+* Create new 4.7.2 .NET Framework blank/empty website
+* Install latest UmbracoCMS 8.1+ Nuget Packages from Nuget.org
```
Comment (marcemarc) [inline Reference/Debugging/index.md]: will this only work if Umbraco is installed via Nuget, apologies I've only skim read  the links...

File: Reference/Debugging/index.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+## Enabling SourceLink in Visual Studio 2019
+* Verify that VS Debugging option **SourceLink** checked & **Just My Code** option is unchecked
+
+![Visual Studio 2019 Debug Settings for SourceLink](Images/VS19-enable-sourcelink.png)
+
+## What is SourceLink ?
+You can find details from Microsoft about SourceLink here:
+
+https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/sourcelink<br/>
```
Comment (marcemarc) [inline Reference/Debugging/index.md]: maybe these are better in a bulleted list - with MD links saying where they go... 

eg

- Microsoft Docs
- Scott Hanselman blog post on Sourcelink
- SourceLink on GitHub

File: Reference/Debugging/index.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+https://www.hanselman.com/blog/ExploringNETCoresSourceLinkSteppingIntoTheSourceCodeOfNuGetPackagesYouDontOwn.aspx<br/>
+https://github.com/dotnet/sourcelink
+
+## Verifying this works
+* Create new 4.7.2 .NET Framework blank/empty website
+* Install latest UmbracoCMS 8.1+ Nuget Packages from Nuget.org
+* Create an IUserComposer or similar code in your new site/SLN that you want to F11/Step Into
+* Prompt should appear & the original source code file is fetched directly from GitHub
```
Comment (marcemarc) [inline Reference/Debugging/index.md]: can we add a screengrab of prompt?

File: Reference/Debugging/index.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+![Visual Studio 2019 Debug Settings for SourceLink](Images/VS19-enable-sourcelink.png)
+
+## What is SourceLink ?
+You can find details from Microsoft about SourceLink here:
+
+https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/sourcelink<br/>
+https://www.hanselman.com/blog/ExploringNETCoresSourceLinkSteppingIntoTheSourceCodeOfNuGetPackagesYouDontOwn.aspx<br/>
+https://github.com/dotnet/sourcelink
+
+## Verifying this works
```
Comment (marcemarc) [inline Reference/Debugging/index.md]: instead of verifying this works how about:

Working with SourceLink

File: Reference/Debugging/index.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+* Create new 4.7.2 .NET Framework blank/empty website
+* Install latest UmbracoCMS 8.1+ Nuget Packages from Nuget.org
+* Create an IUserComposer or similar code in your new site/SLN that you want to F11/Step Into
+* Prompt should appear & the original source code file is fetched directly from GitHub
+* How far can you `F11` aka `Step Into` & go down the rabbit hole of source code?
```
Comment (marcemarc) [inline Reference/Debugging/index.md]: go down the rabbit hole of the Umbraco CMS source code?

File: Reference/Debugging/index.md (hunk 3 above)
Comment (warrenbuckley) [reply Reference/Debugging/index.md]: I need to check & verify @marcemarc 90% sure it is

File: Reference/Debugging/index.md (hunk 6 above)
Comment (warrenbuckley) [reply Reference/Debugging/index.md]: Yep good idea 👍

File: Reference/Debugging/index.md (hunk 7 above)
Comment (warrenbuckley) [reply Reference/Debugging/index.md]:
```suggestion
## Working with SourceLink
```

File: Reference/Debugging/index.md (hunk 8 above)
Comment (warrenbuckley) [reply Reference/Debugging/index.md]:
```suggestion
* How far can you `F11` aka `Step Into` & go down the rabbit hole of the Umbraco CMS source code?
```

File: Reference/Debugging/index.md (hunk 3 above)
Comment (warrenbuckley) [reply Reference/Debugging/index.md]:
```suggestion
## Enabling SourceLink in Visual Studio 2017 & 2019
```

File: Reference/Debugging/index.md (hunk 1 above)
Comment (warrenbuckley) [reply Reference/Debugging/index.md]:
```suggestion
# Debugging with SourceLink
```

File: Reference/Debugging/index.md (hunk 2 above)
Comment (warrenbuckley) [reply Reference/Debugging/index.md]:
```suggestion
Microsoft & Visual Studio have introduced a new debugging technology called 'SourceLink' that enables source code debugging of certain .NET assemblies from NuGet - in version Umbraco 8.1+ this feature has been enabled to allow developers to step into the native Umbraco CMS source code.
```

File: Reference/Debugging/index.md (hunk 4 above)
Comment (warrenbuckley) [reply Reference/Debugging/index.md]: Yes as the vanilla ZIP package the bin folder does not contain the PDBs which has the magic SourceLink meta data inside them

**Comment by warrenbuckley** [comment]
Think I done all your suggestions @marcemarc so let me know :)

File: Reference/Debugging/index.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
+---
+versionFrom: 8.1.0
+---
+
+# Debugging with SourceLink
+Microsoft & Visual Studio have introduced a new debugging technology called 'SourceLink' that enables source code debugging of certain .NET assemblies from NuGet - in version Umbraco 8.1+ this feature has been enabled to allow developers to step into the native Umbraco CMS source code.
```
Comment (sofietoft) [inline Reference/Debugging/index.md]:
```suggestion
Microsoft & Visual Studio have introduced a new debugging technology called 'SourceLink' that enables source code debugging of certain .NET assemblies from NuGet. In version Umbraco 8.1+ this feature has been enabled to allow developers to step into the native Umbraco CMS source code.
```

**Pull request merged**